video/avi_decoder.cpp

```diff
@@ -385,6 +398,21 @@ void AVIDecoder::readStreamName(uint32 size) {
 	}
 }
 
+void AVIDecoder::readPalette8(uint32 size) {
+	if (size < 768) {
+		warning("AVI palette8 is too small (%d, expected >= 768)", size);
+		skipChunk(size);
+		return;
+	}
+
+	if (!_lastAddedTrack || size != 1024) {
+		skipChunk(size);
+	} else {
+		AVIVideoTrack *vidTrack = dynamic_cast<AVIVideoTrack *>(_lastAddedTrack);
```

Comment:
can also be a `reinterpret_cast` as there is no null check done here - and thus no runtime impact.

Reply:
Thanks. Made me realise I *should* have a null check, so I did that instead.